src/lib/y2firewall/widgets/interfaces_table.rb

```diff
@@ -0,0 +1,98 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "cwm"
+require "cwm/table"
+require "y2firewall/ui_state"
+
+module Y2Firewall
+  module Widgets
+    # A table with all {Y2Firewall::Firewalld::Zone}s.
+    class InterfacesTable < ::CWM::Table
+      DEFAULT_ZONE_NAME = "default".freeze
+
+      # @!attribute [r] interfaces
+      #   @return [Array<Hash>] Interfaces
```

Comment:
When a Hash is really an ad-hoc class, like it is here for the interface, I like to make a subclass of Hash just for documentation purposes, like here:
declaration: https://github.com/yast/yast-yast2/blob/e4ab10035ba30f739175009344872c4062c26db6/library/systemd/src/lib/yast2/systemd/unit_prop_map.rb#L8
reference: https://github.com/yast/yast-yast2/blob/e4ab10035ba30f739175009344872c4062c26db6/library/systemd/src/lib/yast2/systemd/socket.rb#L70

So I think we should declare Y2Firewall::Firewalld::Interface

Reply:
I agree. @teclator please, could you adapt `known_interfaces` to work with proper objects while I am extracting the popup and the `UIState` classes?

Reply:
@mvidner @imobachgs please see https://github.com/yast/yast-yast2/pull/832